config/bin_version_strings.cfg

```diff
@@ -330,29 +331,37 @@ glibc;;lgpl;"GNU\ C\ Library\ development\ release\ version\ [0-9](\.[0-9]+)+?$"
-libexpat;;unknown;"expat_[0-9]\.[0-9]+\.[0-9]+$";"sed -r 's/expat_([0-9](\.[0-9]+)+?)$/expat:\1/'";
+libexpat;;mit;"expat_[0-9]\.[0-9]+\.[0-9]+$";"sed -r 's/expat_([0-9](\.[0-9]+)+?)$/expat:\1/'";
 libgcrypt;;lgplv2.1;"[Ll]ibgcrypt\ [0-9](\.[0-9]+)+?\ ";"sed -r 's/[Ll]ibgcrypt\ ([0-9](\.[0-9]+)+?)\ .*/libgcrypt:\1/'";
 libgcrypt;;lgplv2.1;"[Ll]ibgcrypt\ [0-9](\.[0-9]+)+?$";"sed -r 's/[Ll]ibgcrypt\ ([0-9](\.[0-9]+)+?)$/libgcrypt:\1/'";
 libgcrypt;;lgplv2.1;"[Ll]ibgcrypt\ [0-9](\.[0-9]+)+?(-[a-z]+)?$";"sed -r 's/[Ll]ibgcrypt\ ([0-9](\.[0-9]+)+?(-[a-z]+)?)$/libgcrypt:\1/'";
 libgcrypt;;lgplv2.1;"^This\ is\ Libgcrypt\ [0-9]\.([0-9]+)+?\ -\ The\ GNU\ Crypto\ Library$";"sed -r 's/This\ is\ [Ll]ibgcrypt\ ([0-9](\.[0-9]+)+?)\ .*/libgcrypt:\1/'";
-libjpeg;;bsd-style;"Independent\ JPEG\ Group.s\ CJPEG,\ version\ [0-9][a-z]\ \ [0-9]+-[A-Za-z]+-[0-9]+$";"sed -r 's/Independent\ JPEG\ Group.s\ CJPEG,\ version\ ([0-9][a-z])\ .*/libjpeg:\1/'";
-libjpeg-turbo;;bsd-style;"^libjpeg-turbo\ version\ [0-9](\.[0-9]+)+?\ \(build\ [0-9]+\)$";"sed -r 's/libjpeg-turbo\ version\ ([0-9](\.[0-9]+)+?)\ .*/libgjpeg-turbo:\1/'";
-liblzma;;public-domain;"liblzma\ [0-9](\.[0-9]+)+?$";"sed -r 's/liblzma\ ([0-9](\.[0-9]+)+?)$/liblzma:\1/'";
+libgmp;multi_grep;unknown;'"^GNU MP assertion failed: %s$"&&"^[0-9](\.[0-9]+)?+$"';"sed -r 's/([0-9](\.[0-9]+)+?)$/gmplib:\1/'";
 libgpiod;;lgpl;"^gpiofind\ \(libgpiod\)\ [0-9](\.[0-9]+)+?$";"sed -r 's/gpiofind\ \(libgpiod\)\ [0-9](\.[0-9]+)+?)$/libgpiod:\1/'";
 libgpiod;;lgpl;"^gpioinfo\ \(libgpiod\)\ [0-9](\.[0-9]+)+?$";"sed -r 's/gpioinfo\ \(libgpiod\)\ [0-9](\.[0-9]+)+?)$/libgpiod:\1/'";
 libgpiod;;lgpl;"^gpiomon\ \(libgpiod\)\ [0-9](\.[0-9]+)+?$";"sed -r 's/gpiomon\ \(libgpiod\)\ [0-9](\.[0-9]+)+?)$/libgpiod:\1/'";
 libsoup;;lgpl;"^libsoup\/[0-9](\.[0-9]+)+?$";"sed -r 's/^libsoup\/([0-9](\.[0-9]+)+?)$/libsoup:\1/'";
-libmicrohttpd.so.12;strict;lgplv2.1;"^[0-9]\.[0-9](\.[0-9]+)+?$";"sed -r 's/^([0-9]\.[0-9](\.[0-9]+)+?)/gnu:libmicrohttpd:\1/'";
 libharfbuzz.so.0;strict;mit;"^[0-9]\.[0-9](\.[0-9]+)+?$";"sed -r 's/^([0-9]\.[0-9](\.[0-9]+)+?)/harfbuzz_project:harfbuzz:\1/'";
-libnss3;multi_grep;unknown;'"nss_DumpCertificateCacheInfo"&&"^[0-9](\.[0-9]+)?+$"';"sed -r 's/([0-9](\.[0-9]+)+?)$/libnss3:\1/'";
+libidn2;multi_grep;unknown;'"^IDN2_UNKNOWN$"&&"^[0-9](\.[0-9]+)?+$"';"sed -r 's/([0-9](\.[0-9]+)+?)$/libidn2:\1/'";
+libjpeg;;bsd-style;"Independent\ JPEG\ Group.s\ CJPEG,\ version\ [0-9][a-z]\ \ [0-9]+-[A-Za-z]+-[0-9]+$";"sed -r 's/Independent\ JPEG\ Group.s\ CJPEG,\ version\ ([0-9][a-z])\ .*/libjpeg:\1/'";
+libjpeg-turbo;;bsd-style;"^libjpeg-turbo\ version\ [0-9](\.[0-9]+)+?\ \(build\ [0-9]+\)$";"sed -r 's/libjpeg-turbo\ version\ ([0-9](\.[0-9]+)+?)\ .*/libgjpeg-turbo:\1/'";
+libjson-c;multi_grep;mit;'"^INTERNAL ERROR: _json_c_visit"&&"^[0-9](\.[0-9]+)?+$"';"sed -r 's/([0-9](\.[0-9]+)+?)$/json-c:\1/'";
+liblz4.so;strict;unknown;"^[0-9](\.[0-9]+)+?$";"sed -r 's/([0-9](\.[0-9]+)+?)$/liblz4:\1/'";
```

Comment:
do we have some kind of identfier that we can use for multi_grep? We are now trying to avoid strict identifiers.
Same on 353 and 355

Reply:
If you mean liblz4 here, no unfortunately there was no unique identifier. With strings I only get the function names and these are not unique. I can provide the library to you if you want to check if there is a better idea.

Reply:
can we search for multiple non unique strings? We can combine more than two search strings

Reply:
I know, priority for me would be to avoid false positives, I have no idea what to use then, here is the library:
[liblzma.so.5.2.5.zip](https://github.com/e-m-b-a/emba/files/13888763/liblzma.so.5.2.5.zip)

Reply:
hmmm, we could you the library SONAME:

```
 0x000000000000000e (SONAME)             Library soname: [liblzma.so.5]


└─$ strings ~/Downloads/liblzma.so.5.2.5 | grep liblzma
liblzma.so.5
```

and one function name like "lzma_version_string"
The third identifier ist then the version itself

What do you mean?

Reply:
Oops, I mixed liblzma and liblz4, I will search for a solution, just give me some time

Reply:
@m-1-k-3 : I currently have problems that the multi grep is not working any more for me. I thought i messed up my environment, but even after a clean reinstall it does not work. Can you quickly confirm that multi-grep is still working for you in the latest emba version?

Reply:
not good ... not good
Will check here

Reply:
indeed ... it is broken. Will fix it

Reply:
Could you do a check if it is also working for you again?

Reply:
@m-1-k-3 : Working again and I just verified that my additions are working. Do we have open points to fix ?